### ISSUE TITLE
Portus WEB does not show repos

### ISSUE THREAD

**Issue by felixPG** [SUSE/Portus#1775]
Hi All:

I run portus with mysql, configured with a private docker registry. 
All are running ok, I can login in docker with a user created in portus, I can Pull and Push images from the user namespace, etc... but I DONT SEE ANYTHING in the portus webui, like images that I push or a notification on dashboard that say me user X pull image Y. 

What happend with this?

Regards

Felix

**Comment by Vad1mo** [comment]
you should check the `NOTIFICATIONS_ENDPOINTS` part of your registry config.
Also if you have compose make sure that the registry can communicate with Portus.


Can you post the registry config here?

```
      REGISTRY_NOTIFICATIONS_ENDPOINTS: >
        - name: portus
          url: http://${MACHINE_FQDN}:3000/v2/webhooks/events
          timeout: 2000ms
          threshold: 5
          backoff: 1s
```

**Comment by felixPG** [comment]
@Vad1mo thanks for your reply, but...

Tag and push an image:

```
docker tag busybox 192.168.133.8:8080/blue/busybox
docker push 192.168.133.8:8080/blue/busybox
The push refers to a repository [192.168.133.8:8080/blue/busybox]
0314be9edf00: Pushed
latest: digest: sha256:0b29c5182b3fa4c81ffac17702e362532730be4f7f762f32d1b98e0af56fab5e size: 527

```
...**meanwhile in Portus**:

```
Started GET "/v2/token?account=cityneco&scope=repository%3Ablue%2Fbusybox%3Apush%2Cpull&service=192.168.133.8%3A8080" for 172.30.0.1 at 2018-04-10 15:41:51 +0000
Processing by Api::V2::TokensController#show as JSON
  Parameters: {"account"=>"cityneco", "scope"=>"repository:blue/busybox:push,pull", "service"=>"192.168.133.8:8080"}
Completed 200 OK in 220ms (Views: 0.7ms | ActiveRecord: 9.0ms)
Started POST "/v2/webhooks/events" for 172.30.0.1 at 2018-04-10 15:41:52 +0000
Processing by Api::V2::EventsController#create as JSON
Completed 202 Accepted in 1ms (ActiveRecord: 0.0ms)
Started POST "/v2/webhooks/events" for 172.30.0.1 at 2018-04-10 15:41:52 +0000
Processing by Api::V2::EventsController#create as JSON
Completed 202 Accepted in 1ms (ActiveRecord: 0.0ms)
Started POST "/v2/webhooks/events" for 172.30.0.1 at 2018-04-10 15:41:52 +0000
Processing by Api::V2::EventsController#create as JSON
Completed 202 Accepted in 1ms (ActiveRecord: 0.0ms)
Started POST "/v2/webhooks/events" for 172.30.0.1 at 2018-04-10 15:41:52 +0000
Processing by Api::V2::EventsController#create as JSON
Completed 202 Accepted in 1ms (ActiveRecord: 0.0ms)
Started POST "/v2/webhooks/events" for 172.30.0.1 at 2018-04-10 15:41:52 +0000
Processing by Api::V2::EventsController#create as JSON
Completed 202 Accepted in 29ms (ActiveRecord: 4.0ms)
```
So events arrive at portus from registry without problems.

Here my config:
```
notifications:
  endpoints:
    - name: portus
      url: http://192.168.133.8/v2/webhooks/events
      timeout: 500ms
      threshold: 5
      backoff: 1s
```
**192.168.133.8**: Portus
**192.168.133.8:8080**: Docker private registry

**Comment by felixPG** [comment]
@Vad1mo WTF with this?
I restart Portus, and then I see images that I had pushed 1 hour ago

![portus](https://user-images.githubusercontent.com/30522864/38568020-db78e67a-3ce7-11e8-9a16-d7bf86aba155.png)

.**.But the busybox images of the example above does not appear**!!!!

**Comment by Vad1mo** [comment]
Do you have Registry sync enabled it is it default set to initial.

The background job processes the events. Check in the DB for any not
processed event. Or the registry could not deliver the event after retry

**Comment by felixPG** [comment]
Thanks for you reply @Vad1mo. I enable the background proccess rising the issue that I describe here:
https://github.com/SUSE/Portus/issues/1777

So, I cant syncronize the registry on bootstrap process.

**Comment by mimiz** [comment]
Hi, 

Actually, I configured also notification, and have the exactly same problem, I mean, portus receiving data, but does not save anything in database.

```
Processing by Api::V2::TokensController#show as JSON
  Parameters: {"account"=>"remi", "scope"=>"repository:httpd:pull", "service"=>"registry.rgo.ovh"}
Completed 200 OK in 209ms (Views: 0.2ms | ActiveRecord: 39.4ms)
Started GET "/v2/token?account=remi&scope=repository%3Aremi%2Falpine%3Apush%2Cpull&scope=repository%3Aalpine%3Apull&service=my.fqdn" for 10.244.2.78 at 2018-05-10 03:39:40 +0000
Processing by Api::V2::TokensController#show as JSON
  Parameters: {"account"=>"remi", "scope"=>"repository:alpine:pull", "service"=>"my.fqdn"}
Completed 200 OK in 260ms (Views: 0.2ms | ActiveRecord: 41.7ms)
Started POST "/v2/webhooks/events" for 10.244.2.78 at 2018-05-10 03:39:44 +0000
Processing by Api::V2::EventsController#create as JSON
Completed 202 Accepted in 0ms (ActiveRecord: 0.0ms)
Started POST "/v2/webhooks/events" for 10.244.2.78 at 2018-05-10 03:39:44 +0000
Processing by Api::V2::EventsController#create as JSON
Completed 202 Accepted in 0ms (ActiveRecord: 0.0ms)
Started POST "/v2/webhooks/events" for 10.244.2.78 at 2018-05-10 03:39:44 +0000
Processing by Api::V2::EventsController#create as JSON
Completed 202 Accepted in 0ms (ActiveRecord: 0.0ms)
Started POST "/v2/webhooks/events" for 10.244.2.78 at 2018-05-10 03:39:44 +0000
Processing by Api::V2::EventsController#create as JSON
Completed 202 Accepted in 0ms (ActiveRecord: 0.0ms)
Started POST "/v2/webhooks/events" for 10.244.2.78 at 2018-05-10 03:39:45 +0000
Processing by Api::V2::EventsController#create as JSON
Completed 202 Accepted in 8ms (ActiveRecord: 5.1ms)
```

**Comment by felixPG** [comment]
@mimiz please, do you resolve the issue? Portus in my opinion is very unstable, is not ready for production environment. I imteresting in add clair to portus, but portus is not a trusted app for production!

**Comment by mimiz** [comment]
Hi, @felixPG I did not resolve this issue. But I still try to find a way to solve it ;)

And in my opinion, Portus is clearly not ready for production. Or maybe, deployed as a suse package, as a single application, but not in a container (or on kubernetes).

The idea behind the project is clearly good, and the team has already made a great job. We need to give them feedback in order to help them understand our problems.

Regards

**Comment by ivoribeiro** [comment]
Hi @felixPG do you solve the problem, i'm having the same issue, the registry is saving the repo but portus dont insert any data on repos table

**Comment by danielclasen** [comment]
I also had the issue that the UI was not showing any of the repos which were already present in the registry. I thought it was related to the issue #760 where I also commented and described the behavior. But I was able to get the sync from registry to portus running the moment I created the namespaces manually. So all the repos were structured like this project/artifact[:tag] because we used harbor as UI in the past. Apparently the project part has to match a namespace in Portus and this is not synced automatically. I don't know if I overlooked that in the docs or if it is just missing but it took me ages to figure that out.

**TL;DR** Add the namespaces in portus manually if you have namespaces/projects in your docker registry already.

**Comment by mssola** [comment]
> Add the namespaces in portus manually if you have namespaces/projects in your docker registry already.

Yeah. Unfortunately, if you are coming from a full registry, Portus will not automatically create namespaces for you. We need to investigate this in the next release, but automatically creating namespaces raises some questions like: who owns these namespaces? As I said, we want to think about it during the next release, because this is a common pitfall unfortunately...

As for the people that push an image and then it doesn't appear on Portus, could you explain how have you deployed Portus? Because all this (as hinted by @Vad1mo) looks like either a badly configured Docker registry, or missing configuration on the Portus side. One common pitfall is to simply run the Portus docker image, but without adding also the background process as described [here](http://port.us.org/docs/background.html), and showcased in some of our [examples](https://github.com/SUSE/Portus/blob/master/examples/compose/docker-compose.insecure.yml#L36).

If you are still lost on this issue, could you paste some info like:

- What does `portusctl rake portus:info` print if you are on `2.3` or `portusctl exec rake portus:info` if you are on `head`.
- How have you deployed Portus? Are you using any of the examples as a guide? If possible, could you paste some configuration files without sensible data (e.g. docker-compose.yml) ?

**Comment by fabriziogaliano** [comment]
Hi there,

at the moment i can only login on registry pull/push images i've a namespace created on portus but nothing it's showed, the notification are properly POST ono portus:

below the docker compose

`version: '3'

networks:
  default:
    external:
      name: private

services:
   portus_web:
      #image: opensuse/portus:2.3
      image: opensuse/portus:head

      container_name: portus_web

      extra_hosts:
        - portus.xxx.xxx:192.168.0.3
        - registry.xxx.xxx:192.168.0.4    

      volumes:
        - "/docker/management/certbot/letsencrypt:/certificates:ro"

      environment:
        PORTUS_SECRET_KEY_BASE: "xxxxxxxxxxxxxxxxxxxxxxxxxxxx"
        PORTUS_KEY_PATH: "/certificates/live/portus.xxx.xxx/privkey.pem"
        PORTUS_PASSWORD: "xxxxxxxx"
        RAILS_SERVE_STATIC_FILES: "true"
        PORTUS_MACHINE_FQDN_VALUE: "portus.xxx.xxx"

        #SSL
        #PORTUS_CHECK_SSL_USAGE_ENABLED: false
        PORTUS_PUMA_TLS_KEY: "/certificates/live/portus.xxx.xxx/privkey.pem"
        PORTUS_PUMA_TLS_CERT: "/certificates/live/portus.xxx.xxx/fullchain.pem"

        #Mysql connection
        PORTUS_DB_ADAPTER: "mysql2"
        PORTUS_DB_HOST: "portus_mysql"
        PORTUS_DB_PORT: "3306"
        PORTUS_DB_DATABASE: "portus"
        PORTUS_DB_USERNAME: "portus"
        PORTUS_DB_PASSWORD: "xxx"
        PORTUS_DB_POOL: "5"

      ports:
         - "3000:3000"

      restart: always

   portus_registry:
     image: library/registry:2.6

     container_name: portus_registry

      extra_hosts:
        - portus.xxx.xxx:192.168.0.3
        - registry.xxx.xxx:192.168.0.4 

     environment:
       # Authentication
       REGISTRY_AUTH_TOKEN_REALM: "https://portus.xxx.xxx/v2/token"
       REGISTRY_AUTH_TOKEN_SERVICE: "registry.xxx.xxx"
       REGISTRY_AUTH_TOKEN_ISSUER: "portus.xxx.xxx"
       REGISTRY_AUTH_TOKEN_ROOTCERTBUNDLE: "/certificates/live/portus.xxx.xxx/fullchain.pem"
       REGISTRY_HTTP_SECRET: "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"

       REGISTRY_STORAGE_DELETE_ENABLED: "true"
       REGISTRY_LOG_LEVEL: "warn"

       # Portus endpoint
       REGISTRY_NOTIFICATIONS_ENDPOINTS: >
         - name: portus
           url: https://portus.xxx.xxx/v2/webhooks/events
           timeout: 2000ms
           threshold: 5
           backoff: 1s

     volumes:
       - "/docker/management/certbot/letsencrypt:/certificates:ro"
       - "/var/lib/registry:/var/lib/registry"

     ports:
       - 5000:5000

     restart: always

   portus_mysql:
     image: mysql:5.7.22

     restart: always

     container_name: portus_mysql

     environment:
       MYSQL_ROOT_PASSWORD: "xxxxx"
       MYSQL_DATABASE: "portus"
       MYSQL_USER: "portus"
       MYSQL_PASSWORD: "xxxxxx"

     volumes:
       - "/var/lib/mysql/portus:/var/lib/mysql`

**Comment by danielclasen** [comment]
Hey @fabriziogaliano 
it seems that you are missing the portus background container. You need to run another instance of portus which will keep the portus db in sync with the registry. Have a look in that example compose: https://github.com/SUSE/Portus/blob/master/examples/compose/docker-compose.yml
The stuff under the `services.background:` key should be interesting for you ;)

Cheers,
Daniel

**Comment by fabriziogaliano** [comment]
@danielclasen  
Thank you very much! also i've added this command to mysql and for me everything, finally, it's working fine! 

command: ["--character-set-server=utf8", "--collation-server=utf8_unicode_ci"]

YEAH!

**Comment by ajaysheoran2323** [comment]
seems like nobody was able to resolve this...

**Comment by harmeetsinghghotra** [comment]
I am facing the exact issue, I am getting notifications from docker registry and here are the portus logs:

portus_1      | Started POST "/v2/webhooks/events" for 172.25.0.1 at 2019-09-19 16:16:43 +0000
portus_1      | Processing by Api::V2::EventsController#create as JSON
portus_1      | Completed 202 Accepted in 0ms
portus_1      | Started POST "/v2/webhooks/events" for 172.25.0.1 at 2019-09-19 16:16:43 +0000
portus_1      | Processing by Api::V2::EventsController#create as JSON
portus_1      | Completed 202 Accepted in 0ms
portus_1      | Started POST "/v2/webhooks/events" for 172.25.0.1 at 2019-09-19 16:16:43 +0000
portus_1      | Processing by Api::V2::EventsController#create as JSON
portus_1      | Completed 202 Accepted in 0ms (ActiveRecord: 0.0ms)
portus_1      | Started POST "/v2/webhooks/events" for 172.25.0.1 at 2019-09-19 16:16:43 +0000
portus_1      | Processing by Api::V2::EventsController#create as JSON
portus_1      | Completed 202 Accepted in 0ms (ActiveRecord: 0.0ms)
portus_1      | Started POST "/v2/webhooks/events" for 172.25.0.1 at 2019-09-19 16:16:43 +0000
portus_1      | Processing by Api::V2::EventsController#create as JSON
portus_1      | Completed 202 Accepted in 0ms

Also when I start portus and background containers I get the following log as well but the containers start without any issues:

portus_1      | /usr/bin/bundle:23:in `load': cannot load such file -- /usr/lib64/ruby/gems/2.6.0/gems/bundler-1.16.4/exe/bundle (LoadError)
background_1  | /usr/bin/bundle:23:in `load': cannot load such file -- /usr/lib64/ruby/gems/2.6.0/gems/bundler-1.16.4/exe/bundle (LoadError)

And here's one more error that randomly pops up I have no idea why:


and finally, here's my docker-compose.yml:

version: "2"
services:
  portus:
    image: opensuse/portus:head
    environment:
      - PORTUS_MACHINE_FQDN_VALUE=portus.xxx.xxx
      - PORTUS_DB_HOST=db
      - PORTUS_DB_DATABASE=portus_production
      - PORTUS_DB_PASSWORD=xxxxxxxxxxxx
      - PORTUS_DB_POOL=5
      - PORTUS_SECRET_KEY_BASE=xxxxxxxxxxxx
      - PORTUS_KEY_PATH=/certificates/portus.key
      - PORTUS_PASSWORD=xxxxxxxxxxxx
      - PORTUS_PUMA_TLS_KEY=/certificates/portus.key
      - PORTUS_PUMA_TLS_CERT=/certificates/portus.crt
      - PORTUS_CHECK_SSL_USAGE_ENABLED=true
      - RAILS_SERVE_STATIC_FILES=true
    ports:
      - 3000:3000
    links:
      - db
    volumes:
      - ./secrets:/certificates:ro
  background:
    image: opensuse/portus:head
    depends_on:
      - portus
      - db
    environment:
      - PORTUS_MACHINE_FQDN_VALUE=portus.xxx.xxx
      - CCONFIG_PREFIX=PORTUS
      - PORTUS_DB_HOST=db
      - PORTUS_DB_DATABASE=portus_production
      - PORTUS_DB_PASSWORD=xxxxxxxxxxxx
      - PORTUS_DB_POOL=5
      - PORTUS_SECRET_KEY_BASE=xxxxxxxxxxxx
      - PORTUS_KEY_PATH=/certificates/portus.key
      - PORTUS_PASSWORD=xxxxxxxxxxxx
      - PORTUS_BACKGROUND=true
    links:
      - db
    volumes:
      - ./secrets:/certificates:ro
  db:
    image: library/mariadb:10.0.23
    command: mysqld --character-set-server=utf8 --collation-server=utf8_unicode_ci --init-connect='SET NAMES UTF8;' --innodb-flush-log-at-trx-commit=0
    environment:
      - MYSQL_DATABASE=portus_production
      - MYSQL_ROOT_PASSWORD=xxxxxxxxxxxx
    volumes:
      - /var/lib/portus/mariadb:/var/lib/mysql

**Comment by harmeetsinghghotra** [comment]
Am I missing anything? Can anybody help? I am struggling to get this setup working from the last four days. After initial catalog sync from the background process, nothing is done by the background process no matter what I do.

**Comment by kylegoetz** [comment]
@ajaysheoran2323 actually it was resolved: they asked for explanation of how the system was set up, one person shared his docker-compose.yml, they told him what was wrong, and it immediately resolved the problem.

**Comment by kylegoetz** [comment]
All who still have this issue, the registry and Portus are only kept in sync with the background process. However, by default, the background process only syncs one time, and that's when you first spin up the container. In order to have it sync regularly (every minute or quicker, it appears), you need to add some stuff to your docker-compose.yml file under the background service's environment.

I explain it here in my issue: https://github.com/SUSE/Portus/issues/2275
